File: compress.js
const compress = (s) => {
  let str = ''
  
  for(let i = 0; i < s.length; i++){
    let c = s[i]
    let num = 0
    while(s[i] === c){
      num++;
      i++;
    }
    str += num === 1 ? c :  num + c
    i--;
  }
  
  return str
};

File: compress.py
def compress(s):
  num = 0
  result = ''
  base = s[0]
  
  for c in s:
    if base == c:
      num += 1
    else:
      result += str(num) + base if num != 1 else base
      base = c
      num = 1


  result += str(num) + base if num != 1 else base

  return result
      
  